src/mon/ConnectionTracker2.cc

```diff
@@ -0,0 +1,193 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2019 Red Hat
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#include "ConnectionTracker2.h"
+
+ConnectionReport *ConnectionTracker2::reports(int p)
+{
+  auto i = peer_reports.find(p);
+  if (i == peer_reports.end()) {
+    ceph_assert(p != rank);
+    auto[j,k] = peer_reports.insert(std::pair<int,ConnectionReport*>(p,new ConnectionReport()));
+    i = j;
+  }
+  return i->second;
+}
+
+const ConnectionReport *ConnectionTracker2::reports(int p) const
+{
+  auto i = peer_reports.find(p);
+  if (i == peer_reports.end()) {
+    return NULL;
+  }
+  return i->second;
+}
+
+void ConnectionTracker2::receive_peer_report(const ConnectionReport& report)
+{
+  if (report.rank == rank) return;
+  ConnectionReport& existing = *reports(report.rank);
+  if (report.epoch > existing.epoch ||
+      (report.epoch == existing.epoch &&
+       report.epoch_version > existing.epoch_version)) {
+    existing = report;
+  }
+}
+
+bool ConnectionTracker2::increase_epoch(epoch_t e)
+{
+  if (e > epoch) {
+    my_reports.epoch_version = version = 0;
+    my_reports.epoch = epoch = e;
+    return true;
+  }
+  return false;
+}
+
+void ConnectionTracker2::increase_version()
+{
+  ++version;
+  my_reports.epoch_version = version;
+  /*  if ((version % 10) == 0 ) { // TODO: make this configurable?
+    owner->persist_connectivity_scores();
+    } */
```

Comment:
?

Reply:
Relics of redoing the implementation; it's gone in the final.